### PR TITLE
Create standalone Jenkinsfile for building docker images

### PR DESCRIPTION
**Pull request by cm421**
### Intent

Moves linux docker image into its own, standalone Jenkinsfile.

### Approach

With the new matrix pipelines for the Jenkinsfiles, having this image building and the product builds in the same `Jenkinsfile.linux` file caused a `General error during class generation: Method too large` error. The only way to fix that is to break up the code.

I've also added several OS-specific environment variables to simplify the pipeline structure. For example in Ubuntu Jammy:
```
ENV PACKAGE_TYPE='DEB'
ENV PACKAGE_EXT = 'deb'
ENV OS_X86_64_LABEL = 'amd64'
```

I don't _love_ having an uppercase and lowercase version of `DEB`, but this meant not having to add a `sed` line in the pipeline which I felt was cleaner. But open to opinions.

We could also add windows image building into this file too, if we'd like. 

### Automated Tests

None. Build system change

### QA Notes

None. Build system change

### Documentation
None. Build system change

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

File: docker/Jenkinsfile.docker [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+              label "linux && ${arch}"
+            }
+
+            environment {
+              GITHUB_LOGIN = credentials('github-rstudio-jenkins')
+            }
+
+            steps {
+              script {
+                versionFlower = readFile(file: 'version/RELEASE').replaceAll(" ", "-").toLowerCase().trim()
```
[the remark this answers is not in the document]
Comment (cm421) [reply docker/Jenkinsfile.docker:69]: I was thinking the same thing. I'll add it

[the remark this answers is not in the document]
Comment (AndrewMcClain) [reply docker/Jenkinsfile.docker:69]: Couldn't this be wrapped into getVersion, since it's related?

File: docker/Jenkinsfile.docker [hunk 2]
```diff
@@ -21,6 +34,19 @@ pipeline {
   }
 
   stages {
+    stage('Version') {
```
Comment (MariaSemple) [inline docker/Jenkinsfile.docker:37]: We need a checkout step at the commit hash before this

**Review by MariaSemple — APPROVED** [review]
Just one minor thing, otherwise LGTM!

File: jenkins/Jenkinsfile.docker [hunk 3]
```diff
@@ -14,13 +27,41 @@ pipeline {
...
+                   branches: [[name: "${params.COMMIT_HASH}"]],
+                   extensions: [],
+                   userRemoteConfigs: [[credentialsId: 'github-rstudio-jenkins', url: "${GIT_URL}"]]])
+       }
+    }
+
+    stage('Version') {
+      steps {
+        script {
+          sh 'printenv'
```
Comment (zachhannum) [inline jenkins/Jenkinsfile.docker:55]: nit: We should remove this in case it prints out sensitive information.

**Review by zachhannum — APPROVED** [review]
One minor nit, otherwise LGTM!

**Pull request merged**